lib/system/io.nim

```diff
@@ -612,6 +612,25 @@ proc writeFile*(filename, content: string) {.tags: [WriteIOEffect], benign.} =
   else:
     sysFatal(IOError, "cannot open: " & filename)
 
+
+proc readLines*(filename: string, n = 1.Natural): seq[TaintedString] =
```

Comment:
Please don't add this to ``io.nim``, there is got to be a better way.

Reply:
I thought what can be a better way. The only idea that might be questionally better, is to make this example work in compile time:
```nim
static:
  var file = open(file, "copying.txt"):
  for s in lines(file):
    echo s
  close(file)
```
with some hackery in vmops.nim (ptr File <-> IntLitNode casting) and vm callbacks I think it is possible. Is this something you would prefer? the only other option I can think is a separate `readLines` proc just like in this PR.

Reply:
Well if it's only used by the test case, move it to the test case.

Reply:
Introduced `readLines` is used at compile time and there is also vm callback added for this proc. 
Comment mentions that it is available at compile time.
It is used to read the `discard "asasdf687687a6sdf7"` header of the file to check if the file needs to be regenerated by nim macro.